.github/workflows/cicd-pipeline.yml

```diff
@@ -110,61 +110,64 @@ jobs:
-          password: ${{ secrets.REGISTRY_PWD }}
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
 
-      - name: Pull deploy image
-        run: docker pull ${{ env.DOCKER_REGISTRY }}/steuerlotse_deployment
-        shell: bash
+    uses: ./.github/workflows/reusable-deploy-otc.yml
+    with:
+      imageTag: ${{ github.run_number }}
+      environment: staging
+    secrets:
+      DOCKER_REGISTRY: ${{ secrets.DOCKER_REGISTRY }}
+      REGISTRY_USER: ${{ secrets.REGISTRY_USER }}
+      REGISTRY_PWD: ${{ secrets.REGISTRY_PWD }}
+      ERICA_SSH_KEY: ${{ secrets.CI_ERICA_SSH_KEY }}
+      ERICA_HOST: ${{ secrets.ERICA_HOST_OTC_STAGING }}
+      CERT_PIN: ${{ secrets.CERT_PIN_OTC_STAGING }}
+      QUEUE_URL: ${{ secrets.QUEUE_URL_OTC_STAGING }}
+      ERICA_DATABASE_URL: ${{ secrets.ERICA_DATABASE_URL_OTC_STAGING }}
+      DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+      ARGOCD_SERVER: ${{ secrets.ARGOCD_SERVER }}
+      ARGOCD_PIPELINE_PASSWORD: ${{ secrets.ARGOCD_PIPELINE_PASSWORD }}
+
+  track-deploy:
+    needs: [deploy-staging-syseleven, deploy-staging-otc]
 
-      - name: Run deployment
-        env:
-          NAMESPACE: erica-staging
-          IMAGE_TAG: ${{ github.run_number }}
-        run: |
-          docker run \
-          -e DOCKER_REGISTRY='${{ env.DOCKER_REGISTRY }}' \
-          -e ERICA_SSH_KEY='${{ secrets.CI_ERICA_SSH_KEY }}' \
-          -e ERICA_HOST='${{ env.ERICA_HOST }}' \
-          -e REGISTRY_PWD='${{ secrets.REGISTRY_PWD }}' \
-          -e CERT_PIN='${{ secrets.CERT_PIN_STAGING }}' \
-          -e QUEUE_URL='${{ secrets.QUEUE_URL_STAGING }}' \
-          -e ERICA_DATABASE_URL='${{ secrets.ERICA_DATABASE_URL_STAGING }}' \
-          -e KUBECONFIG_BASE64='${{ secrets.CI_KUBECONFIG_B64 }}' \
-          -e NAMESPACE='${{ env.NAMESPACE }}' \
-          -e IMAGE_TAG='${{ env.IMAGE_TAG }}' \
-          ${{ env.DOCKER_REGISTRY }}/steuerlotse_deployment erica
-        shell: bash
-
-      - name: Logout of docker container
-        run: docker logout ${{ env.DOCKER_REGISTRY }}
+    runs-on: ubuntu-latest
 
+    steps:
       - name: Report Deployment
         uses: satak/webrequest-action@v1.2.3
-        env:
-          IMAGE_TAG: ${{ github.run_number }}
         with:
           url: ${{ secrets.METRICS_DEPLOYMENT_WEBHOOK_URL }}
           method: POST
-          payload: '{"project": "Erica", "version_identifier": "${{ env.IMAGE_TAG }}", "environment": "staging", "link": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"}'
+          payload: '{"project": "Erica", "version_identifier": "${{ github.run_number }}", "environment": "staging", "link": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"}'
           headers: '{"Authorization": "Token ${{ secrets.METRICS_WEBHOOK_TOKEN }}"}'
 
-  contract-tests:
-    needs: [deploy-staging]
+  contract-tests-syseleven:
+    needs: [deploy-staging-syseleven]
```

Comment:
Why do we not run the contract tests for the queue on OTC?

Reply:
We need to be inside the cluster for that, and I don't want to poke a hole into the cluster just for that purpose. I've set them up as a post-sync job here: https://github.com/digitalservice4germany/erica-infra/blob/main/manifests/overlays/staging/contract-tests-job.yaml.

Still looking for ways to fail the build or at least notify us when they fail though (currently you have to spot it in argo). Maybe once we have this? https://github.com/digitalservice4germany/platform/issues/8